CIScripts/Test/TestConfigurationUtils.ps1

```diff
@@ -322,83 +328,77 @@ function Initialize-TestConfiguration {
 
-            Start-Sleep -s $SleepTimeBetweenChecks
-        }
+            {
+                Test-IsDockerDriverEnabled -Session $Session
+            } | Invoke-UntilSucceeds -Duration 600 -Interval 5 -Precondition $TestProcessRunning
 
-        if ($Res -ne $true) {
+            break
+        }
+        catch {
             if ($i -eq $NRetries) {
                 throw "Docker driver was not enabled."
             } else {
                 Write-Host "Docker driver was not enabled, retrying."
                 Stop-ProcessIfExists -Session $Session -ProcessName "contrail-windows-docker"
             }
-        } else {
-            break;
         }
     }
 
     $HNSTransparentAdapter = Get-RemoteNetAdapterInformation `
             -Session $Session `
-            -AdapterName $TestConfiguration.VHostName
+            -AdapterName $SystemConfig.VHostName
     Wait-RemoteInterfaceIP -Session $Session -ifIndex $HNSTransparentAdapter.ifIndex
-
-    if (!$NoNetwork) {
-        throw "Creating network in Initialize-TestConfiguration is deprecated"
-    }
 }
 
 function Clear-TestConfiguration {
     Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session,
-           [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+           [Parameter(Mandatory = $true)] [SystemConfig] $SystemConfig)
 
     Write-Host "Cleaning up test configuration"
 
     Remove-AllUnusedDockerNetworks -Session $Session
     Disable-AgentService -Session $Session
     Disable-DockerDriver -Session $Session
-    Disable-VRouterExtension -Session $Session -AdapterName $TestConfiguration.AdapterName `
-        -VMSwitchName $TestConfiguration.VMSwitchName -ForwardingExtensionName $TestConfiguration.ForwardingExtensionName
+    Disable-VRouterExtension -Session $Session -SystemConfig $SystemConfig
 }
 
 function New-AgentConfigFile {
-    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session,
-           [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session,
+        [Parameter(Mandatory = $true)] [ControllerConfig] $ControllerConfig,
+        [Parameter(Mandatory = $true)] [SystemConfig] $SystemConfig
+    )
 
     # Gather information about testbed's network adapters
     $HNSTransparentAdapter = Get-RemoteNetAdapterInformation `
             -Session $Session `
-            -AdapterName $TestConfiguration.VHostName
+            -AdapterName $SystemConfig.VHostName
 
     $PhysicalAdapter = Get-RemoteNetAdapterInformation `
             -Session $Session `
-            -AdapterName $TestConfiguration.AdapterName
+            -AdapterName $SystemConfig.AdapterName
 
     # Prepare parameters for script block
-    $ControllerIP = $TestConfiguration.ControllerIP
+    $ControllerIP = $ControllerConfig.Address
     $VHostIfName = $HNSTransparentAdapter.ifName
     $VHostIfIndex = $HNSTransparentAdapter.ifIndex
 
+    # TODO ???
```

Comment:
More explicit explanation could be a better hint for reader.

Reply:
I'll remove this `TODO` – these lines will be removed in #44 anyway